### DISCUSSION TITLE
Startup blocked by missing network access

### DISCUSSION THREAD

**Discussion by dr460nf1r3** [nextcloud/all-in-one#3846] [Questions] [answered]
### Steps to reproduce
1. Use the following [config](https://github.com/garuda-linux/infrastructure-nix/blob/main/docker-compose/all-in-one/docker-compose.yml#L35) to start the master container
2. Start all applications
3. Observe containers not starting

### Expected behavior 

Containers and Nextcloud are starting up correctly.

### Actual behavior 

The Nextcloud container fails to install all necessary components, preventing a startup. 

### Host OS 

NixOS in a nixos-container (basically systemd-nspawn), Nixpkgs commit `f780534ea2d0c12e62607ff254b6b45f46653f7a`

#### Nextcloud AIO version 

`7.7.1`

#### Current channel 

`latest`

#### Other valuable info 

![image](https://github.com/nextcloud/all-in-one/assets/12834713/f112ea5a-38aa-49bd-bc58-f32074b9c539)

Logs of the `nextcloud-aio-nextcloud` container:

~~~
              now              
-------------------------------
 2023-11-30 14:47:25.254788+00
(1 row)

+ '[' -f /dev-dri-group-was-added ']'
++ find /dev -maxdepth 1 -mindepth 1 -name dri
+ '[' -n '' ']'
+ set +x
Installing imagemagick via apk...
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/main: temporary error (try again later)
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/community: temporary error (try again later)
ERROR: unable to select packages:
  imagemagick (no such package):
    required by: world[imagemagick]
The packet imagemagick was not installed!
Enabling Imagick...
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/main: temporary error (try again later)
WARNING: fetching https://dl-cdn.alpinelinux.org/alpine/v3.18/community: temporary error (try again later)
ERROR: unable to select packages:
  .docker-php-ext-enable-deps-20231130.144745:
    masked in: cache
    satisfies: world[.docker-php-ext-enable-deps=20231130.144745]
  binutils (no such package):
    required by: .docker-php-ext-enable-deps-20231130.144745[binutils]
Could not install PHP extension imagick!
Configuring Redis as session handler...
Checking connection to appstore
Curl didn't produce a 200 status, is appstore reachable?
Checking connection to appstore
(goes on like this forever)
~~~

It seems that the network access is straight out not working in all containers started by the mastercontainer:

~~~
[root@docker:/var/garuda/docker-compose-runner/all-in-one]# docker exec -it nextcloud-aio-nextcloud sh
/var/www/html # ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1): 56 data bytes
^C
--- 1.1.1.1 ping statistics ---
22 packets transmitted, 0 packets received, 100% packet loss
/var/www/html # 
~~~

All other containers started via docker-compose, also the master container, have working network access. 
~~~
[root@docker:/var/garuda/docker-compose-runner/all-in-one]# docker exec -it nextcloud-aio-mastercontainer sh
/var/www/docker-aio # ping 1.1.1.1
PING 1.1.1.1 (1.1.1.1): 56 data bytes
64 bytes from 1.1.1.1: seq=0 ttl=56 time=5.503 ms
64 bytes from 1.1.1.1: seq=1 ttl=56 time=5.451 ms
~~~

The whole thing started yesterday after a server reboot, which changed absolutely nothing related to Nextcloud or Docker setup (just IPv6 addresses, which this application doesn't make use of). Previously, it has been working flawlessly. I'm quite clueless about what's going on here right now.

**Comment by szaimen** [comment]
Hi, do you run a pi-hole or vpn on the same server?

**Comment by dr460nf1r3** [comment]
Hey, we have the server joined to a Tailnet (Tailscale VPN), but I already verified that it makes no difference whether it's turned on or off.

**Comment by szaimen** [comment, marked as answer]
I fear I cannot reproduce your problem on my test server so I think it is infrastructure related...
Moving to discussions for now due to that.
